### ISSUE TITLE
iOS - Search - Background behind the transaction thread is blank when swiping to right

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#43478]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.81-8
**Reproducible in staging?:** Y
**Reproducible in production?:** New feature
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Launch New Expensify app.
2. Go to Search.
3. Open any transaction.
4. Swipe the page to the right and do not release the finger.

## Expected Result:
The background behind the transaction thread will show Search page.

## Actual Result:
The background behind the transaction thread is blank.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/ceee9287-0e71-4460-9e0e-8a88a68a3d65

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
@puneetlath I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think this issue might be related to the #collect project.

**Comment by luacmartins** [comment]
I don't think this needs to be a blocker since the feature works and it's a minor UI issue.

**Comment by TheGithubDev** [comment]
### **Proposal**

### **Please re-state the problem that we are trying to solve in this issue:**
The background behind the transaction thread is blank when swiping to the right on iOS devices.

### **What's the root cause of that problem?**
The root cause of this issue is due to the way the navigation stack or screen rendering is managed during the swipe gesture. When swiping to the right, the screen behind the transaction thread is not properly rendered or visible, resulting in a blank background.

### **What changes do you think we should make in order to solve the problem?**
Ensure that the background screen (Search page) is properly rendered and visible during the swipe gesture. This can be achieved by:
### **1. Ensure Proper Rendering of Background Screen:**
- Make sure the Search page is retained in the view hierarchy and is properly rendered when the transaction thread is swiped to the right.
### **2. Adjust Navigation Stack Management:**
- Ensure the navigation stack is managed in a way that the Search page remains visible behind the transaction thread during the swipe gesture.

_Please note that the above explanation outlines only the technical approach I plan to take. I'll submit the finalized solution after getting accepted, as mentioned._

_Thanks!_

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The background is blank when swiping to close the report RHP.

### What is the root cause of that problem?
When we go to the search page, we add both search bottom tab and search central pane. (both have the same url)
https://github.com/Expensify/App/blob/8d11d0b185083a6a813daeb0fe7c0697a5627ef6/src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar/index.tsx#L100-L104

In small screen, we only want to show the search bottom tab, so, to hide the central pane in small screen, we have a custom navigation state logic to hide the search central pane.
https://github.com/Expensify/App/blob/8d11d0b185083a6a813daeb0fe7c0697a5627ef6/src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx#L69-L79

One condition that we need to focus is `isLastRouteSearchRoute`. It checks whether the last route is the search central pane or not.
https://github.com/Expensify/App/blob/8d11d0b185083a6a813daeb0fe7c0697a5627ef6/src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx#L65

However, when we open the report/transaction RHP (or any page), the last route is not search central pane anymore, but the report RHP, thus the search central pane is not hidden anymore. When we swipe to right from the report RHP, the blank screen is the search central pane. Why blank?

It's because we return null if it's a small screen.
https://github.com/Expensify/App/blob/8d11d0b185083a6a813daeb0fe7c0697a5627ef6/src/pages/Search/SearchPage.tsx#L38-L42

### What changes do you think we should make in order to solve the problem?
Always filter out the search central pane from the navigation state to render.
```
if (isSmallScreenWidth) {
    const searchCentralPaneIndex = routes.findIndex(route => route.name === 'CentralPaneNavigator' && route.params && 'screen' in route.params && route.params.screen === SCREENS.SEARCH.CENTRAL_PANE);
    const filteredRoutes = [...routes.slice(0, searchCentralPaneIndex), ...routes.slice(searchCentralPaneIndex + 1)];
    return {
        stateToRender: {
            ...state,
            index: filteredRoutes.length - 1,
            routes: filteredRoutes,
        },
        searchRoute: routes[searchCentralPaneIndex],
    };
}
```

**Comment by luacmartins** [comment]
The RCA and solution make sense to me. Maybe we could use a `.filter` to make the code a bit cleaner though. Otherwise looks good.

**Comment by bernhardoj** [comment]
PR is ready

cc: @eh2077

**Comment by puneetlath** [comment]
@bernhardoj has been paid. @eh2077 bump on the checklist so we can pay you.

**Comment by eh2077** [comment]
### Checklist

- [x] [@eh2077] The PR that introduced the bug has been identified. Link to the PR: _**I don't think there's a specific PR responsible for this issue. It's an edge case that wan't covered by app-navigation**_
- [x] [@eh2077] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x] [@eh2077] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] [@eh2077] Determine if we should create a regression test for this bug. Yes

#### Regression test

1. Open search page
2. Open any report
3. Verify the transaction screen opens
4. (iOS/mWeb iOS) Swipe right and don't release the finger (Android/Android mWeb) Press back button
5. Verify you can see the search page list under the transaction screen

Do we agree 👍 or 👎

**Comment by puneetlath** [comment]
Regression test issue: https://github.com/Expensify/Expensify/issues/409354

All paid. Thanks everyone!
